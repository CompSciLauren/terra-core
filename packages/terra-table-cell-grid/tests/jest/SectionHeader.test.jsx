import React from 'react';
import SectionHeader from '../../src/SectionHeader';

// Snapshot test
it('should render a default section header', () => {
  const shallowComponent = shallow(
    <SectionHeader title="test" />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a section header with isCollpased', () => {
  const shallowComponent = shallow(
    <SectionHeader title="test" isCollapsed />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a section header with isCollapsible', () => {
  const shallowComponent = shallow(
    <SectionHeader title="test" isCollapsible />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a section header with level', () => {
  const shallowComponent = shallow(
    <SectionHeader title="test" level={3} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a section header with metaData', () => {
  const shallowComponent = shallow(
    <SectionHeader title="test" metaData={{}} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});

it('should render a section header with onSelect', () => {
  const mockCallBack = jest.fn();

  const shallowComponent = shallow(
    <SectionHeader title="test" isCollapsible onSelect={mockCallBack} />,
  );
  expect(shallowComponent).toMatchSnapshot();
  shallowComponent.find('.section-content').simulate('click');
  shallowComponent.find('.section-content').simulate('keydown', { nativeEvent: { keyCode: 13 } });
  shallowComponent.find('.section-content').simulate('keydown', { nativeEvent: { keyCode: 32 } });
  expect(mockCallBack.mock.calls.length).toEqual(3);
});

it('should render a section header with refCallback', () => {
  const shallowComponent = shallow(
    <SectionHeader title="test" refCallback={jest.fn()} />,
  );
  expect(shallowComponent).toMatchSnapshot();
});
