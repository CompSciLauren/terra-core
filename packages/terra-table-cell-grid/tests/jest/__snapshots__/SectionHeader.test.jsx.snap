// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default section header 1`] = `
<div
  className="section-header"
>
  <div
    className="section-content"
  >
    <h2
      className="title"
    >
      test
    </h2>
  </div>
</div>
`;

exports[`should render a section header with isCollapsible 1`] = `
<div
  className="section-header"
>
  <div
    aria-expanded={true}
    aria-level={1}
    className="section-content is-collapsible"
    data-row-show-focus="true"
    onBlur={[Function]}
    onMouseDown={[Function]}
    role="heading"
    tabIndex="0"
  >
    <div
      className="start"
    >
      <span
        className="accordion-icon is-open"
      />
    </div>
    <div
      className="fill"
    >
      <h2
        className="title"
      >
        test
      </h2>
    </div>
  </div>
</div>
`;

exports[`should render a section header with isCollpased 1`] = `
<div
  className="section-header"
>
  <div
    className="section-content"
  >
    <h2
      className="title"
    >
      test
    </h2>
  </div>
</div>
`;

exports[`should render a section header with level 1`] = `
<div
  className="section-header"
>
  <div
    className="section-content"
  >
    <h3
      className="title"
    >
      test
    </h3>
  </div>
</div>
`;

exports[`should render a section header with metaData 1`] = `
<div
  className="section-header"
>
  <div
    className="section-content"
  >
    <h2
      className="title"
    >
      test
    </h2>
  </div>
</div>
`;

exports[`should render a section header with onSelect 1`] = `
<div
  className="section-header"
>
  <div
    aria-expanded={true}
    aria-level={1}
    className="section-content is-collapsible"
    data-row-show-focus="true"
    onBlur={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="heading"
    tabIndex="0"
  >
    <div
      className="start"
    >
      <span
        className="accordion-icon is-open"
      />
    </div>
    <div
      className="fill"
    >
      <h2
        className="title"
      >
        test
      </h2>
    </div>
  </div>
</div>
`;

exports[`should render a section header with refCallback 1`] = `
<div
  className="section-header"
>
  <div
    className="section-content"
  >
    <h2
      className="title"
    >
      test
    </h2>
  </div>
</div>
`;
